unity/Assets/Scripts/BaseFPSAgentController.cs

```diff
@@ -1073,16 +1075,39 @@ protected bool IsCollided()
 			return GameObject.FindObjectsOfType<SimObj>();
         }
 
+        public void ResetObjectFilter() {
+            this.simObjFilter = null;
+            actionFinished(true);
+        }
+        public void SetObjectFilter(string[] objectIds) {
+            SimObjPhysics[] simObjects = GameObject.FindObjectsOfType<SimObjPhysics>();
+            simObjFilter = new HashSet<SimObjPhysics>();
+            foreach(var simObj in simObjects) {
+                if(Array.IndexOf(objectIds, simObj.ObjectID) >= 0) {
+                    simObjFilter.Add(simObj);
+                }
+            }
+            actionFinished(true);
+        }
+
         public virtual ObjectMetadata[] generateObjectMetadata()
 		{
-            SimObjPhysics[] visibleSimObjs = VisibleSimObjs(false); // Update visibility for all sim objects for this agent
             HashSet<SimObjPhysics> visibleSimObjsHash = new HashSet<SimObjPhysics>();
-            foreach (SimObjPhysics sop in visibleSimObjs) {
-                visibleSimObjsHash.Add(sop);
+            SimObjPhysics[] simObjects = null;
+            if (this.simObjFilter != null) {
+                foreach (SimObjPhysics sop in this.simObjFilter) {
+                    if (isSimObjVisible(m_Camera, sop)) {
+                        visibleSimObjsHash.Add(sop);
+                    }
+                }
+                simObjects = this.simObjFilter.ToArray();
+            } else {
+                foreach (SimObjPhysics sop in VisibleSimObjs(false)) {
+                    visibleSimObjsHash.Add(sop);
+                }
+                simObjects = GameObject.FindObjectsOfType<SimObjPhysics>();
```

Comment:
At line 1122, below (GitHub does not let me comment there), there is a reference to `excludeObject` which uses an array of ids  `this.excludeObjectIds` which probably we can deprecate and I don't see any other references to.